### PR TITLE
feat: updated version of Channel Table Widget

### LINKED ISSUE #125: unable to select channel in mda widget if is widget loaded before cfg
not sure if this is widgets or napari-micromanager, but i suspect widgets:

if I open the mda widget before loading a system config, I seem to be never able to add a channel later, even after loading a config:

https://user-images.githubusercontent.com/1609449/220792317-39c0a121-00ec-448a-a6d9-57c770e3ebec.mov

if I do it in the reverse order, it's fine:

https://user-images.githubusercontent.com/1609449/220792345-97662413-3d7f-4bff-a0f4-73f06b3dbb47.mov

### PR DESCRIPTION
**Pull request by fdrgsp**
This PR mainly:
 - updates the current `ChannelTable` widget by adding a `QComboBox` that can be used to select which micromanager `ChannelGroup` to use in the `ChannelTable` (video below).

- adds a general `ChannelGroupWidget` which is a simple `QComboBox` that can be used  to keep track/change the micromanager `ChannelGroup` property.

- adds `ChannelTable` widget and `ChannelGroupWidget` to the docs.

- adds new tests for both.

merge after https://github.com/pymmcore-plus/pymmcore-plus/pull/180.
___

https://user-images.githubusercontent.com/70725613/213214341-73d4989d-8907-4ffb-9e9f-2037a862ec5f.mov

fixes #125

### CODE REVIEW

File: src/pymmcore_widgets/_mda/_channel_table_widget.py
Comment (tlambert03) [inline src/pymmcore_widgets/_mda/_channel_table_widget.py:153]: does this attribute even exist anymore?

**Review by tlambert03 — COMMENTED** [review]
i guess my main question here would be: haven't we lost some flexibility?

let's say someone had a configuration that had _two_ config groups that were meaningful as channel groups... and they wanted to have a channel table widget dedicated to each of them.  Wouldn't this prevent that, since now the channel table widget is _obligated_ to follow the current group?

also, i think there are some latent bugs here, since `_channel_group` is accessed in places but isn't actually set anywhere.  I'm going to request that this have complete test coverage before merge

File: src/pymmcore_widgets/_mda/_channel_table_widget.py
Comment (fdrgsp) [reply src/pymmcore_widgets/_mda/_channel_table_widget.py]: > let's say someone had a configuration that had _two_ config groups that were meaningful as channel groups... and they wanted to have a channel table widget dedicated to each of them. Wouldn't this prevent that, since now the channel table widget is _obligated_ to follow the current group?

Yes, that's true! What about if we keep both versions? one that is more general (the current one, which we can call something like `GeneralChannelTable`) and this one (`ChannelTable`) which is more similar to micromanager...or we can implement it only in `napari-micromanager` if you think it is better!

> also, i think there are some latent bugs here, since `_channel_group` is accessed in places but isn't actually set anywhere.

That's true, I forgot to remove the variable in few places...Also at the moment the micromanager `ChannelGroup` property is not controlled by the `ChannelTable` widget itself but is only handled by the `ChannelGroupWidget` that I have added in this PR (`_channel_group_widget.py`).

About testing this widget, I added some tests to check that everything worked as I wanted - `test_channel_table_widget.py`.

File: src/pymmcore_widgets/_channel_group_widget.py
Comment (tlambert03) [inline src/pymmcore_widgets/_channel_group_widget.py:27]: make keyword only

File: src/pymmcore_widgets/_mda/_channel_table_widget.py
Comment (tlambert03) [inline src/pymmcore_widgets/_mda/_channel_table_widget.py:50]: move kwarg only

File: src/pymmcore_widgets/_mda/_channel_table_widget.py
Comment (tlambert03) [inline src/pymmcore_widgets/_mda/_channel_table_widget.py:75]: remove if not used internally

**Comment by tlambert03** [comment]
can you update the original description here to clarify the motivation for this pr?  could be a bullet point list of what problem it solves, what wasn't possible  before what is possible now.

File: src/pymmcore_widgets/_mda/_channel_table_widget.py
Comment (tlambert03) [inline src/pymmcore_widgets/_mda/_channel_table_widget.py:61]: 1 is reserved for DecorationRole... see https://doc.qt.io/qt-6/qt.html#ItemDataRole-enum

use a UserRole plus some offset

**Review by tlambert03 — APPROVED** [review]
good to go after changing the data role number

File: src/pymmcore_widgets/_mda/_channel_table_widget.py
Comment (fdrgsp) [reply src/pymmcore_widgets/_mda/_channel_table_widget.py:61]: is it ok to do that?

`CH_GROUP_ROLE = Qt.UserRole + 1`

and then to set the item data:

`channel_combo.setItemData(0, _channel_group, self.CH_GROUP_ROLE)`

The index depends on the number of items in the combobox...

**Pull request merged**
